File: openapi-client/cpp-tiny/lib/Models/SendMailAdv.cpp


#include "SendMailAdv.h"

using namespace Tiny;

SendMailAdv::SendMailAdv()
{
	subject = std::string();
	body = std::string();
	from = EmailAddressTypes();
	to = EmailAddressesTypes();
	replyto = EmailAddressesTypes();
	cc = EmailAddressesTypes();
	bcc = EmailAddressesTypes();
	attachments = std::list<MailAttachment>();
	id = long(0);
}

SendMailAdv::SendMailAdv(std::string jsonString)
{
	this->fromJson(jsonString);
}

SendMailAdv::~SendMailAdv()
{

}

void
SendMailAdv::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *subjectKey = "subject";

    if(object.has_key(subjectKey))
    {
        bourne::json value = object[subjectKey];



        jsonToValue(&subject, value, "std::string");


    }

    const char *bodyKey = "body";

    if(object.has_key(bodyKey))
    {
        bourne::json value = object[bodyKey];



        jsonToValue(&body, value, "std::string");


    }

    const char *fromKey = "from";

    if(object.has_key(fromKey))
    {
        bourne::json value = object[fromKey];




        EmailAddressTypes* obj = &from;
		obj->fromJson(value.dump());

    }

    const char *toKey = "to";

    if(object.has_key(toKey))
    {
        bourne::json value = object[toKey];




        EmailAddressesTypes* obj = &to;
		obj->fromJson(value.dump());

    }

    const char *replytoKey = "replyto";

    if(object.has_key(replytoKey))
    {
        bourne::json value = object[replytoKey];




        EmailAddressesTypes* obj = &replyto;
		obj->fromJson(value.dump());

    }

    const char *ccKey = "cc";

    if(object.has_key(ccKey))
    {
        bourne::json value = object[ccKey];




        EmailAddressesTypes* obj = &cc;
		obj->fromJson(value.dump());

    }

    const char *bccKey = "bcc";

    if(object.has_key(bccKey))
    {
        bourne::json value = object[bccKey];




        EmailAddressesTypes* obj = &bcc;
		obj->fromJson(value.dump());

    }

    const char *attachmentsKey = "attachments";

    if(object.has_key(attachmentsKey))
    {
        bourne::json value = object[attachmentsKey];


        std::list<MailAttachment> attachments_list;
        MailAttachment element;
        for(auto& var : value.array_range())
        {


            element.fromJson(var.dump());

            attachments_list.push_back(element);
        }
        attachments = attachments_list;


    }

    const char *idKey = "id";

    if(object.has_key(idKey))
    {
        bourne::json value = object[idKey];



        jsonToValue(&id, value, "long");


    }


}

bourne::json
SendMailAdv::toJson()
{
    bourne::json object = bourne::json::object();





    object["subject"] = getSubject();






    object["body"] = getBody();







	object["from"] = getFrom().toJson();






	object["to"] = getTo().toJson();






	object["replyto"] = getReplyto().toJson();






	object["cc"] = getCc().toJson();






	object["bcc"] = getBcc().toJson();




    std::list<MailAttachment> attachments_list = getAttachments();
    bourne::json attachments_arr = bourne::json::array();

    for(auto& var : attachments_list)
    {
        MailAttachment obj = var;
        attachments_arr.append(obj.toJson());
    }
    object["attachments"] = attachments_arr;







    object["id"] = getId();



    return object;

}

std::string
SendMailAdv::getSubject()
{
	return subject;
}

void
SendMailAdv::setSubject(std::string subject)
{
	this->subject = subject;
}

std::string
SendMailAdv::getBody()
{
	return body;
}

void
SendMailAdv::setBody(std::string body)
{
	this->body = body;
}

EmailAddressTypes
SendMailAdv::getFrom()
{
	return from;
}

void
SendMailAdv::setFrom(EmailAddressTypes from)
{
	this->from = from;
}

EmailAddressesTypes
SendMailAdv::getTo()
{
	return to;
}

void
SendMailAdv::setTo(EmailAddressesTypes to)
{
	this->to = to;
}

EmailAddressesTypes
SendMailAdv::getReplyto()
{
	return replyto;
}

void
SendMailAdv::setReplyto(EmailAddressesTypes replyto)
{
	this->replyto = replyto;
}

EmailAddressesTypes
SendMailAdv::getCc()
{
	return cc;
}

void
SendMailAdv::setCc(EmailAddressesTypes cc)
{
	this->cc = cc;
}

EmailAddressesTypes
SendMailAdv::getBcc()
{
	return bcc;
}

void
SendMailAdv::setBcc(EmailAddressesTypes bcc)
{
	this->bcc = bcc;
}

std::list<MailAttachment>
SendMailAdv::getAttachments()
{
	return attachments;
}

void
SendMailAdv::setAttachments(std::list<MailAttachment> attachments)
{
	this->attachments = attachments;
}

long
SendMailAdv::getId()
{
	return id;
}

void
SendMailAdv::setId(long id)
{
	this->id = id;
}

File: swagger-client/micronaut/src/main/java/io/swagger/model/SendMailAdv.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.model.EmailAddressTypes;
import io.swagger.model.EmailAddressesTypes;
import io.swagger.model.MailAttachment;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Details for an Email
 */
@Schema(description = "Details for an Email")
@Validated
@Introspected
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.MicronautCodegen", date = "2025-10-06T16:15:25.436344464-04:00[America/New_York]")
public class SendMailAdv   {
  @JsonProperty("subject")
  private String subject = null;

  @JsonProperty("body")
  private String body = null;

  @JsonProperty("from")
  private EmailAddressTypes from = null;

  @JsonProperty("to")
  private EmailAddressesTypes to = null;

  @JsonProperty("replyto")
  private EmailAddressesTypes replyto = null;

  @JsonProperty("cc")
  private EmailAddressesTypes cc = null;

  @JsonProperty("bcc")
  private EmailAddressesTypes bcc = null;

  @JsonProperty("attachments")
  @Valid
  private List<MailAttachment> attachments = null;

  @JsonProperty("id")
  private Long id = null;

  public SendMailAdv subject(String subject) {
    this.subject = subject;
    return this;
  }

  /**
   * The subject or title of the email
   * @return subject
  **/
  @Schema(example = "Your Package has been Delivered!", required = true, description = "The subject or title of the email")
  @NotNull

  public String getSubject() {
    return subject;
  }

  public void setSubject(String subject) {
    this.subject = subject;
  }

  public SendMailAdv body(String body) {
    this.body = body;
    return this;
  }

  /**
   * The main email contents.
   * @return body
  **/
  @Schema(example = "The package you ordered on 2021-01-23 has been delivered. If the package is broken into many pieces, please blaim someone else.", required = true, description = "The main email contents.")
  @NotNull

  public String getBody() {
    return body;
  }

  public void setBody(String body) {
    this.body = body;
  }

  public SendMailAdv from(EmailAddressTypes from) {
    this.from = from;
    return this;
  }

  /**
   * Get from
   * @return from
  **/
  @Schema(required = true, description = "")
  @NotNull

  @Valid
  public EmailAddressTypes getFrom() {
    return from;
  }

  public void setFrom(EmailAddressTypes from) {
    this.from = from;
  }

  public SendMailAdv to(EmailAddressesTypes to) {
    this.to = to;
    return this;
  }

  /**
   * Get to
   * @return to
  **/
  @Schema(required = true, description = "")
  @NotNull

  @Valid
  public EmailAddressesTypes getTo() {
    return to;
  }

  public void setTo(EmailAddressesTypes to) {
    this.to = to;
  }

  public SendMailAdv replyto(EmailAddressesTypes replyto) {
    this.replyto = replyto;
    return this;
  }

  /**
   * Get replyto
   * @return replyto
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public EmailAddressesTypes getReplyto() {
    return replyto;
  }

  public void setReplyto(EmailAddressesTypes replyto) {
    this.replyto = replyto;
  }

  public SendMailAdv cc(EmailAddressesTypes cc) {
    this.cc = cc;
    return this;
  }

  /**
   * Get cc
   * @return cc
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public EmailAddressesTypes getCc() {
    return cc;
  }

  public void setCc(EmailAddressesTypes cc) {
    this.cc = cc;
  }

  public SendMailAdv bcc(EmailAddressesTypes bcc) {
    this.bcc = bcc;
    return this;
  }

  /**
   * Get bcc
   * @return bcc
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public EmailAddressesTypes getBcc() {
    return bcc;
  }

  public void setBcc(EmailAddressesTypes bcc) {
    this.bcc = bcc;
  }

  public SendMailAdv attachments(List<MailAttachment> attachments) {
    this.attachments = attachments;
    return this;
  }

  public SendMailAdv addAttachmentsItem(MailAttachment attachmentsItem) {
    if (this.attachments == null) {
      this.attachments = new ArrayList<MailAttachment>();
    }
    this.attachments.add(attachmentsItem);
    return this;
  }

  /**
   * (optional) File attachments to include in the email.  The file contents must be base64 encoded!
   * @return attachments
  **/
  @Schema(example = "[         {             \"filename\": \"text.txt\",             \"data\": \"base64_encoded_contents\"         } ]", description = "(optional) File attachments to include in the email.  The file contents must be base64 encoded!")
  @NotNull
  @Valid
  public List<MailAttachment> getAttachments() {
    return attachments;
  }

  public void setAttachments(List<MailAttachment> attachments) {
    this.attachments = attachments;
  }

  public SendMailAdv id(Long id) {
    this.id = id;
    return this;
  }

  /**
   * (optional)  ID of the Mail order within our system to use as the Mail Account.
   * @return id
  **/
  @Schema(example = "5000", description = "(optional)  ID of the Mail order within our system to use as the Mail Account.")
  @NotNull

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    SendMailAdv sendMailAdv = (SendMailAdv) o;
    return Objects.equals(this.subject, sendMailAdv.subject) &&
        Objects.equals(this.body, sendMailAdv.body) &&
        Objects.equals(this.from, sendMailAdv.from) &&
        Objects.equals(this.to, sendMailAdv.to) &&
        Objects.equals(this.replyto, sendMailAdv.replyto) &&
        Objects.equals(this.cc, sendMailAdv.cc) &&
        Objects.equals(this.bcc, sendMailAdv.bcc) &&
        Objects.equals(this.attachments, sendMailAdv.attachments) &&
        Objects.equals(this.id, sendMailAdv.id);
  }

  @Override
  public int hashCode() {
    return Objects.hash(subject, body, from, to, replyto, cc, bcc, attachments, id);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class SendMailAdv {\n");
    
    sb.append("    subject: ").append(toIndentedString(subject)).append("\n");
    sb.append("    body: ").append(toIndentedString(body)).append("\n");
    sb.append("    from: ").append(toIndentedString(from)).append("\n");
    sb.append("    to: ").append(toIndentedString(to)).append("\n");
    sb.append("    replyto: ").append(toIndentedString(replyto)).append("\n");
    sb.append("    cc: ").append(toIndentedString(cc)).append("\n");
    sb.append("    bcc: ").append(toIndentedString(bcc)).append("\n");
    sb.append("    attachments: ").append(toIndentedString(attachments)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
